src/components/GenerativeDirectAnswer.tsx

```diff
@@ -0,0 +1,167 @@
+import { 
+  GenerativeDirectAnswerResponse, 
+  useSearchActions, 
+  useSearchState, 
+  Result
+} from '@yext/search-headless-react';
+import { useComposedCssClasses } from '../hooks';
+import { executeGenerativeDirectAnswer } from '../utils/search-operations';
+import ReactMarkdown from "react-markdown";
+import React from 'react';
+
+/**
+ * The CSS class interface used for {@link GenerativeDirectAnswer}.
+ *
+ * @public
+ */
+export interface GenerativeDirectAnswerCssClasses {
+  container?: string,
+  header?: string,
+  answerText?: string,
+  divider?: string,
+  citationsContainer?: string,
+  citation?: string,
+  citationTitle?: string,
+  citationSnippet?: string
+}
+
+const builtInCssClasses: Readonly<GenerativeDirectAnswerCssClasses> = {
+  container: 'p-6 border border-gray-200 rounded-lg shadow-sm',
+  header: 'text-xl',
+  answerText: 'mt-4',
+  divider: 'border-b border-gray-200 w-full pb-6 mb-6',
+  citationsContainer: 'mt-4 flex overflow-x-auto gap-4',
+  citation: 'p-4 border border-gray-200 rounded-lg shadow-sm bg-slate-100 flex flex-col grow-0 shrink-0 basis-64 text-sm text-neutral',
+  citationTitle: 'font-bold',
+  citationSnippet: 'line-clamp-2 text-ellipsis break-words'
+};
+
+/**
+ * Props for {@link GenerativeDirectAnswer}.
+ *
+ * @public
+ */
+export interface GenerativeDirectAnswerProps {
+  /** CSS classes for customizing the component styling. */
+  customCssClasses?: GenerativeDirectAnswerCssClasses,
+  /** The header for the answer section of the generative direct answer. */
+  answerHeader?: string | JSX.Element
+  /** The header for the citations section of the generative direct answer. */
+  citationsHeader?: string | JSX.Element
+}
+
+/**
+ * Displays the AI generated answer of a generative direct answer.
+ *
+ * @public
+ *
+ * @param props - {@link GenerativeDirectAnswerProps}
+ * @returns A React element for the generative direct answer, or null if there is no generated answer
+ */
+export function GenerativeDirectAnswer({
+  customCssClasses,
+  answerHeader,
+  citationsHeader
+}: GenerativeDirectAnswerProps): JSX.Element | null {
+  const cssClasses = useComposedCssClasses(builtInCssClasses, customCssClasses);
+
+  const universalResults = useSearchState(state => state.universal);
+  const verticalResults = useSearchState(state => state.vertical);
+
+  const searchResults: Result[] | undefined = React.useMemo(() => {
+    if (universalResults) {
+      return universalResults.verticals?.flatMap(v => v.results);
+    } else if (verticalResults) {
+      return verticalResults.results;
+    }
+  }, [universalResults, verticalResults]);
+
+  const searchActions = useSearchActions();
+  const gdaResponse = useSearchState(state => state.generativeDirectAnswer?.response);
+  const isLoading = useSearchState(state => state.generativeDirectAnswer?.isLoading);
+  
+  React.useEffect(() => {
+    if (!searchResults?.length) {
+      return;
+    }
+    executeGenerativeDirectAnswer(searchActions);
```

Comment:
if there's an error for the request, the old gda results would still be saved and rendered right? I think we should clear the result on failure

Reply:
would you want me to update search-headless so that the generative direct answer state is cleared if there is an error executing gda? 
When adding the gda endpoint, I was copying the flow of the other endpoints like executeUniversalQuery, in which the universal state was not cleared upon error. I tried forcing executeUniversalQuery to return `Promise.reject()`, but the old universal result was still rendered on the test page.

Reply:
I think updating in headless make sense. If it handles state update on success, it should handle on failure as well.